Modules/CTP/src/RawDataReaderCheck.cxx

```diff
@@ -81,55 +135,89 @@ Quality RawDataReaderCheck::check(std::map<std::string, std::shared_ptr<MonitorO
+        if (inputRates && !relativeRates) {
           fHistPrev = (TH1F*)fHistInputPrevious->Clone();
-        if (mo->getName() == "classes")
+        }
+        if (inputRates && relativeRates) {
+          fHistPrev = (TH1F*)fHistInputRatioPrevious->Clone();
+        }
+        if (!inputRates && !relativeRates) {
           fHistPrev = (TH1F*)fHistClassesPrevious->Clone();
-        fHistDifference->Scale(1. / 60);
-        fHistPrev->Scale(1. / 60);
-        fHistDifference->Add(fHistPrev, -1);
+        }
+        if (!inputRates && relativeRates) {
+          fHistPrev = (TH1F*)fHistClassRatioPrevious->Clone();
+        }
+        if (!relativeRates) {
+          fHistDifference->Scale(1. / (mCycleDuration * mFraction));
+          fHistPrev->Scale(1. / (mCycleDuration * mFraction));
+        }
+        fHistDifference->Add(fHistPrev, -1); // Calculate relative difference w.r.t. rate in previous cycle
         fHistDifference->Divide(fHistPrev);
 
-        for (size_t i = 1; i < fHistDifference->GetXaxis()->GetNbins() + 1; i++) {
-          if (TMath::Abs(fHistDifference->GetBinContent(i)) > 0.15)
-            vIndexBad.push_back(i);
-          else if (TMath::Abs(fHistDifference->GetBinContent(i)) > 0.1)
-            vIndexMedium.push_back(i);
+        float thrBad = mThresholdRateBad;
+        float thrMedium = mThresholdRateMedium;
+        if (relativeRates) {
+          thrBad = mThresholdRateRatioBad;
+          thrMedium = mThresholdRateRatioMedium;
         }
 
-        if (mo->getName() == "inputs") {
-          fHistInputPrevious = (TH1F*)h->Clone();
-          inputCycleCounter++;
+        for (size_t i = 1; i < fHistDifference->GetXaxis()->GetNbins() + 1; i++) { // Check how many imputs/classes changed more than a theshold value
+          if (TMath::Abs(fHistDifference->GetBinContent(i)) > thrBad) {
+            vIndexBad.push_back(i);
+          } else if (TMath::Abs(fHistDifference->GetBinContent(i)) > thrMedium) {
+            vIndexMedium.push_back(i);
+          }
         }
-        if (mo->getName() == "classes") {
-          fHistClassesPrevious = (TH1F*)h->Clone();
-          classCycleCounter++;
+        if (vIndexBad.size() > 0) {
+          result = Quality::Bad;
+        } else if (vIndexMedium.size() > 0) {
+          result = Quality::Medium;
+        } else {
+          result = Quality::Good;
         }
       }
-      if (vIndexBad.size() > 0)
-        result = Quality::Bad;
-      else if (vIndexMedium.size() > 0)
-        result = Quality::Medium;
-      else
-        result = Quality::Good;
+
+      if (inputRates && !relativeRates) {
+        fHistInputPrevious = (TH1F*)h->Clone();
+      }
+      if (inputRates && relativeRates) {
+        fHistInputRatioPrevious = (TH1F*)h->Clone();
+      }
+      if (!inputRates && !relativeRates) {
+        fHistClassesPrevious = (TH1F*)h->Clone();
+      }
+      if (!inputRates && relativeRates) {
+        fHistClassRatioPrevious = (TH1F*)h->Clone();
```

Comment:
for all of those, you are not deleting any object that could be at this pointer before the new `Clone()`. This looks like a memory leak.